src/uucore/src/lib/features/encoding.rs

```diff
@@ -42,138 +30,115 @@ pub enum Format {
-                Base64Url => b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789=_-",
-                Base32Hex => b"0123456789ABCDEFGHIJKLMNOPQRSTUV=",
-                Base16 => b"0123456789ABCDEF",
-                Base2Lsbf => b"01",
-                Base2Msbf => b"01",
-                Z85 => b"0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ.-:+=^!/*?&<>()[]{}@%$#",
-            },
-        }
-    }
+        output.extend(string.as_bytes());
 
-    #[must_use]
-    pub fn line_wrap(mut self, wrap: usize) -> Self {
-        self.line_wrap = wrap;
-        self
+        Ok(())
     }
+}
 
-    #[must_use]
-    pub fn ignore_garbage(mut self, ignore: bool) -> Self {
-        self.ignore_garbage = ignore;
-        self
-    }
+impl SupportsFastEncode for Encoding {
+    // Adapted from `encode_append` in the "data-encoding" crate
+    fn encode_to_vec_deque(&self, input: &[u8], output: &mut VecDeque<u8>) -> UResult<()> {
+        let output_len = output.len();
 
-    pub fn decode(&mut self) -> DecodeResult {
-        let mut buf = vec![];
-        self.input.read_to_end(&mut buf)?;
-        if self.ignore_garbage {
-            buf.retain(|c| self.alphabet.contains(c));
-        } else {
-            buf.retain(|&c| c != b'\r' && c != b'\n');
-        };
-        decode(self.format, &buf)
-    }
+        output.resize(output_len + self.encode_len(input.len()), 0_u8);
 
-    pub fn encode(&mut self) -> Result<String, EncodeError> {
-        let mut buf: Vec<u8> = vec![];
-        match self.input.read_to_end(&mut buf) {
-            Ok(_) => encode(self.format, buf.as_slice()),
-            Err(_) => Err(EncodeError::InvalidInput),
-        }
+        let make_contiguous_result = output.make_contiguous();
+
+        self.encode_mut(input, &mut (make_contiguous_result[output_len..]));
+
+        Ok(())
     }
 }
 
-// NOTE: this will likely be phased out at some point
-pub fn wrap_print<R: Read>(data: &Data<R>, res: &str) {
-    let stdout = io::stdout();
-    wrap_write(stdout.lock(), data.line_wrap, res).unwrap();
+pub trait SupportsFastDecode {
+    fn decode_into_vec(&self, input: &[u8], output: &mut Vec<u8>) -> UResult<()>;
 }
 
-pub fn wrap_write<W: Write>(mut writer: W, line_wrap: usize, res: &str) -> io::Result<()> {
-    use std::cmp::min;
+impl SupportsFastDecode for ZEightFiveWrapper {
+    fn decode_into_vec(&self, input: &[u8], output: &mut Vec<u8>) -> UResult<()> {
+        if input.first() == Some(&b'#') {
+            return Err(USimpleError::new(1_i32, "error: invalid input".to_owned()));
+        }
 
-    if line_wrap == 0 {
-        return write!(writer, "{res}");
-    }
+        // According to the spec we should not accept inputs whose len is not a multiple of 4.
+        // However, the z85 crate implements a padded encoding and accepts such inputs. We have to manually check for them.
+        if input.len() % 4_usize != 0_usize {
```

Comment:
Again, why specify the integer types explicitly? Please write `4` instead of `4_usize`, or document the reason why this is necessary.

Reply:
Will change